lnst/Recipes/ENRT/SRIOVNetnsBridgeRecipe.py

```diff
@@ -0,0 +1,187 @@
+        configuration = super().test_wide_configuration()
+        configuration.test_wide_devices = []
+
+        for i, host in enumerate([host1, host2]):
+            host.run(f"devlink dev eswitch set pci/{host.eth0.bus_info} mode switchdev; sleep 2")
+            host.run(f"echo 1 > /sys/class/net/{host.eth0.name}/device/sriov_numvfs; sleep 5")
+
+            vf_ifname = dict(ifname=f"{host.eth0.name}v0")
+            host.map_device("vf_eth0", vf_ifname)
+
+            host.newns = NetNamespace(f"lnst{i + 1}")
+            host.newns.vf_eth0 = host.vf_eth0
+
+            vf_representor_ifname = dict(ifname="eth0")
+            host.map_device("vf_representor_eth0", vf_representor_ifname)
+
+            host.br0 = BridgeDevice()
+            host.br0.slave_add(host.eth0)
+            host.br0.slave_add(host.vf_representor_eth0)
+
+            for dev in [host.eth0, host.newns.vf_eth0, host.vf_representor_eth0, host.br0]:
+                dev.up()
+
+            host.newns.vf_eth0.ip_add(ipaddress("192.168.101." + str(i + 1) + "/24"))
+            host.newns.vf_eth0.ip_add(ipaddress("fc00::" + str(i + 1) + "/64"))
+
+            configuration.test_wide_devices.append(host.newns.vf_eth0)
+
+        self.wait_tentative_ips(configuration.test_wide_devices)
+
+        return configuration
+
+    def generate_test_wide_description(self, config):
+        desc = super().generate_test_wide_description(config)
+        host1, host2 = self.matched.host1, self.matched.host2
+        for i, host in enumerate([host1, host2]):
+            desc += [
+                f"Configured {host.hostid}.{host.eth0.name}.driver = switchdev\n"
+                f"Created virtual function on {host.hostid}.{host.eth0.name} = {host.newns.vf_eth0.name}\n"
+                f"Created network_namespace on {host.hostid} = {host.newns.name}\n"
+                f"Moved interface {host.newns.vf_eth0.name} from {host.hostid} to {host.hostid}.{host.newns.name}\n"
+                f"Created Linux bridge on {host.hostid} = {host.br0.name}\n"
+                f"Configured {host.hostid}.{host.br0.name}.slaves = {host.eth0.name}, {host.vf_representor_eth0.name}"
+            ]
+        desc += [
+            f"Configured {dev.host.hostid}.{dev.name}.ips = {dev.ips}"
+                    for dev in config.test_wide_devices
+                 ]
+        return desc
+
+    def test_wide_deconfiguration(self, config):
+        """
+        Test wide deconfiguration means deleting the Linux bridge and returning the
+        control over the physical function to base driver.
+        Finally virtual function is deleted.
+        """
+        host1, host2 = self.matched.host1, self.matched.host2
+        for i, host in enumerate([host1, host2]):
+            host.run(f"echo 0 > /sys/class/net/{host.eth0.name}/device/sriov_numvfs; sleep 5")
+            host.run(f"devlink dev eswitch set pci/{host.eth0.bus_info} mode legacy; sleep 2")
```

Comment:
Could you move sleep commands out of the run()? My concern is that if the first and more important part of the command list fails, the sleep would mask the failed return code because sleep would return 0.

Reply:
Fixed